qiskit_experiments/database_service/db_experiment_data.py

```
@@ -1692,3 +1706,107 @@ def __getattr__(self, name: str) -> Any:
         except KeyError:
             # pylint: disable=raise-missing-from
             raise AttributeError("Attribute %s is not defined" % name)
+
+    def _safe_serialize_jobs(self):
+        """Return serializable object for stored jobs"""
+        # Since Job objects are not serializable this removes
+        # them from the jobs dict and returns {job_id: None}
+        # that can be used to retrieve jobs from a service after loading
+        jobs = ThreadSafeOrderedDict()
+        for jid, _ in self._jobs.items():
+            jobs[jid] = None
+        return jobs
+
+    def _safe_serialize_figures(self):
+        """Return serializable object for stored figures"""
+        # Convert any MPL figures into SVG images before serializing
+        figures = ThreadSafeOrderedDict()
+        for name, figure in self._figures.items():
+            if isinstance(figure, pyplot.Figure):
+                figures[name] = plot_to_svg_bytes(figure)
+            else:
+                figures[name] = figure
+        return figures
+
+    def __json_encode__(self):
+        if any(not fut.done() for fut in self._job_futures.values()):
+            LOG.warning(
```

For job I think it's better to raise an error and ask user to cancel first.